flux-typeck/src/constraint_gen.rs

```diff
@@ -29,6 +29,12 @@ pub struct ConstrGen<'a, 'tcx> {
     tag: Tag,
 }
 
+impl ConstrGen<'_, '_> {
+    pub fn span(&self) -> Option<Span> {
+        self.tag.span()
+    }
+}
+
```

Comment:
I'd put this in the main `impl block`

Reply:
Ah good point -- somehow I missed it the first time round!